genid: add page (#5867)
tldr-pages/tldr: pages/linux/genid.md

# genid

> Generate IDs, such as snowflakes, UUIDs, and a new GAID.

> > More information: <https://github.com/bleonard252/genid>

Reviewer: ```suggestion
> More information: <https://github.com/bleonard252/genid>.
```

> - Generate a UUIDv4:

Reviewer: In #5892, we use `v4 UUID` instead of `UUIDv4`. I don't know which is better, but we should unify it. (cc @peterbabic)

Reviewer 2: @marchersimon agreed, but [the official standard](https://tools.ietf.org/html/rfc4122) mentions it in neither short form but rather in the verbose form of `version 1 UUID` or in some cases `UUID version 1`. Maybe we should follow that. What do you think?

Reviewer: I'm not sure if using the long form is necessary, since it's pretty long and everybody knows what `v1` means. Personally I like `UUIDv1` (probably because of `IPv4` and `IPv6`).

`genid uuid`

> - Generate a UUIDv5 with a namespace UUID and name THISISANAME:

Reviewer: ```suggestion
- Generate a UUIDv5 with a namespace UUID and a specific name:
```

> - Generate a UUIDv5 with a namespace UUID and a specific name:

Reviewer 3: I think the verb is `using` because is used for hashing :thinking: 
```suggestion
- Generate a UUIDv5 using a namespace UUID and a specific name:
```

> `genid uuidv5 {{{ce598faa-8dd0-49ee-8525-9e24fff71dca}}} {{THISISANAME}}`

Reviewer: ```suggestion
`genid uuidv5 {{{ce598faa-8dd0-49ee-8525-9e24fff71dca}}} {{name}}`
```

> `genid uuidv5 {{{ce598faa-8dd0-49ee-8525-9e24fff71dca}}} {{THISISANAME}}`

Reviewer: @bl-ue I think you used an older commit for the suggestion 🤔

> `genid uuidv5 {{{ce598faa-8dd0-49ee-8525-9e24fff71dca}}} {{THISISANAME}}`

Reviewer 4: Oops!  I've had this suggestion sitting out for days, and I meant to delete it, but apparently I sent it 😆

> - Generate a Discord Snowflake, without a trailing newline (for scripts):

Reviewer 3: ```suggestion
- Generate a Discord Snowflake, without a trailing newline (useful in shell scripts):
```

Author: I chose to shorten it since the terminal preview looked a bit strange when it overflowed.

`genid --script snowflake`

> - Generate a Generic Anonymous ID with "real ID" tldrpages:

Reviewer: ```suggestion
- Generate a Generic Anonymous ID with a specific "real ID":
```

> `genid gaid {{tldrpages}}`

Reviewer: ```suggestion
`genid gaid {{real_id}}`
```

> - Generate a Snowflake with the epoch set to March 13, 2020:

Reviewer: ```suggestion
- Generate a Snowflake with the epoch set to a specific date:
```

> `genid snowflake --epoch={{1584072000}}`

Reviewer: ```suggestion
`genid snowflake --epoch={{unix_epoch_time}}`
```
